keel-intent/src/main/kotlin/com/netflix/spinnaker/keel/intent/processor/converter/SecurityGroupConverter.kt

```diff
@@ -52,8 +49,31 @@ class SecurityGroupConverter(
           description = spec.description,
           accountName = spec.accountName,
           region = region,
-          vpcId = networkNameToId(networks, "aws", region, spec.vpcName),
-          inboundRules = spec.inboundRules.map { objectMapper.convertValue<MutableMap<String, Any>>(it, ANY_MAP_TYPE) },
+          // TODO rz - do we even want to mess with EC2-classic support?
+          vpcId = clouddriverCache.networkBy(spec.vpcName!!, spec.accountName, region).id,
+          inboundRules = spec.inboundRules.map {
+            when (it) {
+              is ReferenceSecurityGroupRule -> SecurityGroup.SecurityGroupRule(
+                protocol = it.protocol,
+                portRanges = it.portRanges.map { SecurityGroup.SecurityGroupRulePortRange(it.startPort, it.endPort) },
+                securityGroup = SecurityGroup.SecurityGroupRuleReference(
+                  name = it.name,
+                  accountName = spec.accountName,
+                  region = region
+                )
+              )
+              is CrossAccountReferenceSecurityGroupRule -> SecurityGroup.SecurityGroupRule(
+                protocol = it.protocol,
+                portRanges = it.portRanges.map { SecurityGroup.SecurityGroupRulePortRange(it.startPort, it.endPort) },
+                securityGroup = SecurityGroup.SecurityGroupRuleReference(
+                  name = it.name,
+                  accountName = it.account,
+                  region = it.region
+                )
+              )
+              else -> TODO(reason = "${it.javaClass.simpleName} has not been implemented yet")
+            }
+          },
```

Comment:
I'll be working on refactoring `keel-intent` to be cloud-provider specific soon.